sqlmesh/core/snapshot/cache.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import typing as t
+
+from pathlib import Path
+from sqlmesh.core.model.cache import OptimizedQueryCache
+from sqlmesh.core.snapshot.definition import Snapshot, SnapshotId
+from sqlmesh.utils.cache import FileCache
+
+
+class SnapshotCache:
```

Comment:
should this not be in the state sync cache?

Reply:
I think this is pretty independent from the context in which it's used.